mantidproject/mantid
# Fixing direction cosines

**Description of work.**

The algorithms `SaveHKL` and `SaveHKLCW` incorrectly assumes the the unit vector directions of the scattered and reverse incident beams are the direction cosines. I reality, to make these work for `Jana2020` or `WinGX`, the actual direction cosines need to be computed by performing one more step by taking the dot product with the crystallographic directions. This is clearly spelled out in the reference `A. Katrusiak, Zeitschrift für Kristallographie - Crystalline Materials 216, 646 (2001).`

Historically, this has worked with internal TOPAZ reduction scripts that computes it correctly. The unit tests are updated with a TOPAZ example. This has been tested to work with the extinction corrections in `Jana2020` and the absorption correction program in `WinGX`.

Added some updates to the user docs.

![image](https://user-images.githubusercontent.com/13754794/224576137-174b18c6-6a84-497e-b400-22616755ad2f.png)




**To test:**

Run the unit tests to make sure they pass. Optionally test with  `Jana2020` and `WinGX`. They won't work if the direction cosines are wrong.



*There is no associated issue.*





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
